stdlib/public/core/Integers.swift

```diff
@@ -716,6 +716,9 @@ public protocol BinaryInteger :
   ///     let x = Int8(bitPattern: 0b1111_1000)
   ///     // x == -8
   ///     // x.trailingZeroBitCount == 3
+  ///
+  /// If there are no bits equal to 1 in this value's binary representation,
+  /// then `trailingZeroBitCount` is equal to `bitWidth`.
```

Comment:
Maybe "If this value is zero, then `trailingZeroBitCount` is `bitWidth`." ("no bits equal to 1" feels like we're trying to leave some loophole for other ways this could happen, but really we just mean zero.)

Reply:
I was thinking about the wording here. Currently, the protocol imposes no requirements on the actual binary representation of integer values. If we guarantee `(0 as T).trailingZeroBitCount == (0 as T).bitWidth`, then we are adding a requirement about the binary representation of zero--which is fine, but worth remarking.

Reply:
All of these properties are defined in terms of a twos-complement representation. Even if your type secretly does something else internally, the result is what you would get for twos-complement.

Reply:
But `bitWidth` would then be the actual bit width and not that of the two's-complement representation?

Reply:
No, bitWidth should also be the bitWidth of the two’s complement representation.

Reply:
Interesting. I'd been under the impression that a subset of methods (operators, `words`, etc.) specifically called for the two's-complement representation while another subset (`bitWidth` principally) deal with the actual binary representation. Not sure why I'd been thinking that.